ansible_collections/arista/avd/plugins/plugin_utils/eos_validate_state_utils/avdtestbase.py

```diff
@@ -5,280 +5,62 @@
 
 import logging
 from functools import cached_property
-from ipaddress import ip_interface
-from typing import Mapping
+from typing import TYPE_CHECKING, Mapping
 
-from ansible_collections.arista.avd.plugins.plugin_utils.errors import AristaAvdError, AristaAvdMissingVariableError
-from ansible_collections.arista.avd.plugins.plugin_utils.utils import default, get, get_item
+from .mixins import DeviceUtilsMixin, ValidationMixin
+
+if TYPE_CHECKING:
+    from .config_manager import ConfigManager
 
 LOGGER = logging.getLogger(__name__)
-LOGGING_LEVELS = ["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"]
 
 
-class AvdTestBase:
-    """
-    Base class for all AVD eos_validate_state tests.
-    """
+class AvdTestBase(DeviceUtilsMixin, ValidationMixin):
+    """Base class for all AVD eos_validate_state tests."""
 
-    def __init__(self, device_name: str, hostvars: Mapping):
-        """
-        Initialize the AvdTestBase class.
+    def __init__(self, config_manager: ConfigManager) -> None:
+        """Initialize the AvdTestBase class.
 
         Args:
-            device_name (str): The current device name for which the plugin is being run.
-            hostvars (Mapping): A mapping that contains a key for each device with a value of the structured_config.
-                                      When using Ansible, this is the `task_vars['hostvars']` object.
+        ----
+            config_manager (ConfigManager): The ConfigManager instance for the device.
         """
-        self.hostvars = hostvars
-        self.device_name = device_name
-        self.structured_config = self.get_host_structured_config(host=device_name)
+        self.config_manager = config_manager
+
+    @cached_property
+    def device_name(self) -> str:
+        """Return the device name from the ConfigManager instance."""
+        return self.config_manager.device_name
```

Comment:
is it actually needed to be a cached_property? would not 
```suggestion
    self.device_name = self.config_manager.device_name
```

do the same and keep the value ?

Reply:
True. Changed to `@property` in 554e20b10cc2d157c35c92107939d8e54417c893